resources/qml/WelcomePages/AddPrinterByIpContent.qml

```diff
@@ -0,0 +1,229 @@
+
+                text: catalog.i18nc("@label", "Enter the IP address or hostname of your printer on the network.")
+            }
+
+            Item
+            {
+                id: userInputFields
+                height: childrenRect.height
+                width: parent.width
+                anchors.top: explainLabel.bottom
+
+                TextField
+                {
+                    id: hostnameField
+                    anchors.verticalCenter: addPrinterButton.verticalCenter
+                    anchors.left: parent.left
+                    height: addPrinterButton.height
+                    anchors.right: addPrinterButton.left
+                    anchors.margins: UM.Theme.getSize("default_margin").width
+                    font: UM.Theme.getFont("default")
+
+                    validator: RegExpValidator
+                    {
+                        regExp: /[a-fA-F0-9\.\:]*/
+                    }
+
+                    onAccepted: addPrinterButton.clicked()
+                }
+
+                Cura.PrimaryButton
+                {
+                    id: addPrinterButton
+                    anchors.top: parent.top
+                    anchors.right: parent.right
+                    anchors.margins: UM.Theme.getSize("default_margin").width
+                    width: UM.Theme.getSize("action_button").width
+                    fixedWidthMode: true
+
+                    text: catalog.i18nc("@button", "Add")
+                    onClicked:
+                    {
+                        if (hostnameField.text.trim() != "")
+                        {
+                            enabled = false;
+                            UM.OutputDeviceManager.addManualDevice(hostnameField.text, hostnameField.text);
+                        }
+                    }
+
+                    BusyIndicator
+                    {
+                        anchors.fill: parent
+                        running:
+                        {
+                            ! parent.enabled &&
+                            ! addPrinterByIpScreen.hasSentRequest &&
+                            ! addPrinterByIpScreen.haveConnection
+                        }
+                    }
+                }
+            }
+
+            Rectangle
+            {
+                width: parent.width
+                anchors.top: userInputFields.bottom
+                anchors.margins: UM.Theme.getSize("default_margin").width
+
+                Label
+                {
+                    id: waitResponseLabel
+                    anchors.top: parent.top
+                    anchors.margins: UM.Theme.getSize("default_margin").width
+                    font: UM.Theme.getFont("default")
+
+                    visible: { addPrinterByIpScreen.hasSentRequest && ! addPrinterByIpScreen.haveConnection }
+                    text: catalog.i18nc("@label", "The printer at this address has not responded yet.")
+                }
+
+                Rectangle
```

Comment:
Same as above, it;s not setting a color, so change it to item

Reply:
Fixed in 0c8afbf